src/ncp/CMakeLists.txt

```
@@ -85,6 +85,13 @@ set(COMMON_SOURCES
     ncp_uart.cpp
 )
 
+if("OPENTHREAD_ENABLE_NCP_VENDOR_HOOK=1" IN_LIST OT_PRIVATE_DEFINES)
+    if(NOT OPENTHREAD_NCP_VENDOR_HOOK_SOURCE)
+        message(FATAL_ERROR "OPENTHREAD_NCP_VENDOR_HOOK_SOURCE must be defined when OPENTHREAD_ENABLE_NCP_VENDOR_HOOK=1")
+    endif()
+    list(APPEND COMMON_SOURCES ${OPENTHREAD_NCP_VENDOR_HOOK_SOURCE})
+endif()
```

can OPENTHREAD_NCP_VENDOR_HOOK_SOURCE and OT_NCP_VENDOR_HOOK be a single option?
```suggestion
if(OPENTHREAD_NCP_VENDOR_HOOK_SOURCE)
    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_ENABLE_NCP_VENDOR_HOOK=1")
    list(APPEND COMMON_SOURCES ${OPENTHREAD_NCP_VENDOR_HOOK_SOURCE})
endif()
```

---

Yes they can. I will change it and i will make new commits for that in this pull request and the related Zephyr pull request.